evidence/pool.go

```diff
@@ -190,20 +184,78 @@ func (evpool *Pool) MarkEvidenceAsCommitted(height int64, lastBlockTime time.Tim
 	// make a map of committed evidence to remove from the clist
 	blockEvidenceMap := make(map[string]struct{})
 	for _, ev := range evidence {
-		evpool.store.MarkEvidenceAsCommitted(ev)
-		blockEvidenceMap[evMapKey(ev)] = struct{}{}
+		if evpool.IsCommitted(ev) {
```

Comment:
is this even possible?

Reply:
hmm shouldn't be

Reply:
unless malicious nodes vote on an evidence that's already submitted

Reply:
already committed you mean? which should not happen unless 2/3+ are malicious

Reply:
yeah

Reply:
the only other circumstance is if MarkEvidenceAsCommitted is done twice on the same block